test/browser_test.dart

```diff
@@ -0,0 +1,97 @@
+@TestOn('browser')
+
+import 'package:js/js.dart';
+import 'package:node_interop/js.dart';
+import 'package:node_interop/util.dart';
+import 'package:sass/src/node/compile_options.dart';
+import 'package:sass/src/node/legacy/render_options.dart';
+import 'package:sass/src/node/legacy/render_result.dart';
+import 'package:test/test.dart';
+import 'ensure_npm_package.dart';
+import 'package:sass/src/node/compile_result.dart';
+
+@JS()
+external Sass get sass;
+
+@JS()
+class Sass {
+  external NodeCompileResult compileString(String text,
+      [CompileStringOptions? options]);
+  external Promise compileStringAsync(String text,
+      [CompileStringOptions? options]);
+  external NodeCompileResult compile(String path, [CompileOptions? options]);
+  external Promise compileAsync(String path, [CompileOptions? options]);
+  external void render(
+      RenderOptions options, void callback(Error error, RenderResult result));
+  external RenderResult renderSync(RenderOptions options);
+  external String get info;
+}
+
+void main() {
+  setUpAll(ensureNpmPackage);
+
+  test('compileAsync() is not available', () {
+    expect(
+        () => sass.compileAsync('index.scss'),
+        throwsA(predicate((err) =>
+            (err as dynamic).message ==
+            'The compileAsync() method is only available in Node.js.')));
+  });
+
+  test('compile() is not available', () {
+    expect(
+        () => sass.compile('index.scss'),
+        throwsA(predicate((err) =>
+            (err as dynamic).message ==
+            'The compile() method is only available in Node.js.')));
+  });
+
+  test('render() is not available', () {
+    expect(
+        () => sass.render(RenderOptions(), (error, result) {}),
+        throwsA(predicate((err) =>
+            (err as dynamic).message ==
+            'The render() method is only available in Node.js.')));
```

Comment:
This is the only test failing so far with:

```
Expected: throws satisfies function
    Actual: <Closure 'main__closure5'>
     Which: threw JSObject:<TypeError: Instance of 'main___closure': type 'UnknownJavaScriptObject' is not a subtype of type '(Object?, RenderResult?) => void'>
```

I think the noop callback I'm passing here is failing type checks here, but I'm not sure how to fix it. Would appreciate any pointers @jgerigmeyer @nex3

Reply:
You need to call [`allowInterop()`](https://api.dart.dev/stable/2.9.0/dart-js/allowInterop.html) on every Dart callback you pass to JS.

Reply:
Fixed in https://github.com/oddbird/dart-sass/commit/1937affd6fa9fc3965c72222f323f56fb53352d7